java/java-analysis-impl/src/com/intellij/codeInspection/dataFlow/java/ControlFlowAnalyzer.java

```diff
@@ -842,6 +844,12 @@ else if (closure instanceof PsiLambdaExpression) {
             generateBoxingUnboxingInstructionFor(returnValue, LambdaUtil.getFunctionalInterfaceReturnType(lambdaExpression));
           }
         }
+        if (InheritanceUtil.isInheritor(returnValue.getType(), JAVA_UTIL_STREAM_BASE_STREAM)) {
+          addInstruction(new DupInstruction());
+          addInstruction(new UnwrapDerivedVariableInstruction(SpecialField.CONSUMED_STREAM));
+          addInstruction(new EnsureInstruction(new ConsumedStreamProblem(returnValue), RelationType.NE, DfStreamStateType.CONSUMED, null));
```

Comment:
I'm not sure, that it is reasonable for this point, because it is about return statement. But I added this exception to StreamChainInliner and changed com.intellij.codeInspection.dataFlow.java.ControlFlowAnalyzer#finishCall to support this. 

New tests are added (I simplified a bit them):
```java
  private static void test12E(Stream<Object> stream) {
    stream.forEach(System.out::println);
    String text = "noError";
    try {
      <warning descr="Stream has already been linked or consumed">stream</warning>.forEach(System.out::println);
    } catch (IllegalStateException e) {
      text = "IllegalStateException";
    } catch (RuntimeException e) {
      text = "RuntimeException";
    }
    if (<warning descr="Condition 'text.equals(\"IllegalStateException\")' is always 'true'">text.equals("IllegalStateException")</warning>) {
      System.out.println("error");
    }
  }
```